lib/bots/versionbot.ts

```diff
@@ -448,6 +458,50 @@ export class VersionBot extends ProcBot {
 	}
 
 	/**
+	 * Always passes Versionist and AutoMerge status if a no-checks label has been applied.
+	 * This allows manual merging to correctly go ahead.
+	 *
+	 * @param _registration  GithubRegistration object used to register the method
+	 * @param event          ServiceEvent containing the event information ('pull_request' event)
+	 * @returns              A void Promise once execution has finished.
+	 */
+	protected passWithNoChecks = (_registration: GithubRegistration, event: ServiceEvent): Promise<void> => {
+		const pr = event.cookedEvent.data.pull_request;
+		const head = event.cookedEvent.data.pull_request.head;
+		const owner = head.repo.owner.login;
+		const repo = head.repo.name;
+		const prNumber = pr.number;
+
+		this.logger.log(LogLevel.INFO, `Skipping AutoMerge and Versionist checks for ${owner}/${repo}#${prNumber}`);
+
+		// Always set the automerge and versionist status to passing.
+		return Promise.join([
+			this.dispatchToEmitter(this.githubEmitterName, {
+				data: {
+					context: StatusAutoMerge.Context,
+					description: StatusAutoMerge.ManualMerge,
+					owner,
+					repo,
+					sha: head.sha,
+					state: 'success'
+				},
+				method: this.githubApi.repos.createStatus,
+			}),
+			this.dispatchToEmitter(this.githubEmitterName, {
+				data: {
+					context: StatusVersionist.Context,
+					description: StatusVersionist.NoVersioning,
+					owner,
+					repo,
+					sha: head.sha,
+					state: 'success'
+				},
+				method: this.githubApi.repos.createStatus,
+			})
+		]).return();
```

Comment:
Personally, I usually use `.all`, because it's clearer, but also more flexible and easy to use, and I worry about that more than the performance difference, which don't make much difference in a project like this.

As a case in point, this code is silently broken :smiley:. `.join` always knows the exact number of promises it wants, because it doesn't take a dynamically sized array: it takes a fixed number of parameters. With `.all` you can pass an array, or a promise that resolves to an array, and get sensible behaviour, but if you pass it an array to `.join` (as you're doing here) then it doesn't examine it, and instead treats it as a resolved final value and silently doesn't wait on promises within at all.

This code isn't waiting for these operations, and doesn't handle errors from them.

More clearly:

```typescript
// This prints 'done' instantly, plus a warning about unhandled errors
Promise.join([ Promise.delay(10000), Promise.reject(new Error('oh no!')) ])
.then(() => console.log('done'))
```